File: 프로그래머스/unrated/181861. 배열의 원소만큼 추가하기/배열의 원소만큼 추가하기.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public List solution(int[] arr) {
        List<Integer> answer = new ArrayList<>();
        
        for(int i = 0; i < arr.length; i++) {
            for(int j = 0; j < arr[i]; j++) {
                answer.add(arr[i]);
            } // i는 arr의 index이고 길이만큼 for문이 반복된다
              // j는 arr의 i번째 원소만큼 추가된다
        }
        return answer;
    }
}

File: 프로그래머스/0/181861. 배열의 원소만큼 추가하기/배열의 원소만큼 추가하기.js
function solution(arr) {
    let answer = [];
    
    for (let i = 0; i < arr.length; i++) {
        for (let j = 0; j < arr[i]; j++) {
            answer.push(arr[i]);
        } // i는 arr의 index이고 길이만큼 for문이 반복된다
    }     // j는 arr의 i번째 원소만큼 추가된다
    return answer;
}
